crates/containerd-shim-wasmedge/src/instance.rs

```diff
@@ -235,6 +238,12 @@ impl Instance for Wasi {
         let pid = container.pid().unwrap();
         container.start()?;
 
+        // Close the fds now that they have been passed to the container process
+        // so that we don't leak them.
+        stdin.map(|fd| unsafe { close(fd) });
```

Comment:
Is there a way to close the fd without using `unsafe`?

Reply:
[nix](https://docs.rs/nix/latest/nix/index.html)::[unistd](https://docs.rs/nix/latest/nix/unistd/index.html)::[close](https://docs.rs/nix/latest/nix/unistd/fn.close.html#)

Reply:
Ok fixed! I also moved the FD closing up before `start()`, because the process exists already after the `build()` step. Later we can investigate if we can pass the actual FD ownership to the runtime so that we wouldn't need to explicitly close the FDs.